protocol/test/temple-zaps.ts

```diff
@@ -0,0 +1,2056 @@
+import { ethers, network } from "hardhat";
+import { Signer, BigNumber, BigNumberish, BytesLike, Contract, providers } from "ethers";
+import { expect, should } from "chai";
+import axios from 'axios';
+import { blockTimestamp, shouldThrow, toAtto } from "./helpers";
+import { DEPLOYED_CONTRACTS } from '../scripts/deploys/helpers';
+import addresses from "./constants";
+import { ERC20, ERC20__factory, Exposure, Exposure__factory, Faith, Faith__factory, IERC20, IERC20__factory,
+  JoiningFee,
+  JoiningFee__factory,
+  TempleZaps, TempleZaps__factory,
+  TempleERC20Token,
+  TempleERC20Token__factory,
+  TempleStableAMMRouter,
+  TempleStableAMMRouter__factory,
+  Vault, VaultedTemple, VaultedTemple__factory, 
+  VaultProxy, VaultProxy__factory, Vault__factory, 
+  GenericZap__factory, GenericZap, 
+  UniswapV2Pair, UniswapV2Router02NoEth,
+  UniswapV2Router02NoEth__factory, UniswapV2Pair__factory,
+  ICurvePool__factory, ICurvePool, ICurveFactory__factory, IUniswapV2Pair__factory, IBalancerVault__factory, IBalancerVault
+} from "../typechain";
+
+const { WETH, USDC, UNI, FRAX, ETH, OGT, FEI, BNB, FXS, BAL } = addresses.tokens;
+const { BINANCE_ACCOUNT_8, WETH_WHALE, FRAX_WHALE, FXS_WHALE } = addresses.accounts;
+const { ZEROEX_EXCHANGE_PROXY, TEMPLE_STABLE_ROUTER, UNISWAP_V2_ROUTER, CURVE_FACTORY, 
+  TEMPLE_FRAX_PAIR, BALANCER_VAULT } = addresses.contracts;
+const { MULTISIG, TEMPLE, TEMPLE_V2_ROUTER, FAITH, STAKING } = DEPLOYED_CONTRACTS.mainnet;
+
+const ZEROEX_QUOTE_ENDPOINT = 'https://api.0x.org/swap/v1/quote?';
+const defaultTokens = [
+  FRAX, USDC, FEI, USDC, UNI,
+  WETH, FXS, BNB, BAL, ETH
+];
+
+let genericZaps: GenericZap;
+let templeZaps: TempleZaps;
+let vaultProxy: VaultProxy;
+let vault: Vault;
+let templeExposure: Exposure;
+let vaultedTemple: VaultedTemple;
+let templeToken: TempleERC20Token;
+let joiningFee: JoiningFee;
+let fraxToken: ERC20;
+let faith: Faith;
+let templeRouter: TempleStableAMMRouter;
+let owner: Signer;
+let alice: Signer;
+let alan: Signer;
+let binanceSigner: Signer;
+let wethSigner: Signer;
+let fraxSigner: Signer;
+let ownerAddress: string;
+let aliceAddress: string;
+let alanAddress: string;
+let balancerVault: IBalancerVault;
+let poolId: string;
+let bptPoolToken: string;
+
+
+describe.only("Temple Stax Core Zaps", async () => {
+  before(async () => {
+    // as real-time live data is being queried from 0x API, there's the possibility that prices may differ from forked mainnet tests (due to block number)
+    // and hence tokens out (in tests) may differ from real expected values 
+    // use most recent block number in tests (with enough block confirmations)
+    await resetFork(await getLatestBlockNumberWithEnoughConfirmations());
```

Comment:
have we tried running the entire test suite with this? My concern is any tests following this suite will run with forked chain (not necessarily the worst thing but is a performance hit)

Reply:
well before, I had run the entire test suite even with the forked state and remaining tests afterwards finished successfully.
But will double-check and worst case scenario i will rename this test to be last or use a global `after()` hook

Reply:
double-checked and all tests pass locally. also, included an `after()` hook